compiler/luci/pass/src/helpers/LayerInfoMap.cpp

```diff
@@ -22,6 +22,134 @@
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV4OUT:
+    {
+      const auto non_max_sus_v4_out = loco::must_cast<CircleNonMaxSuppressionV4Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v4_out->input());
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV5OUT:
+    {
+      const auto non_max_sus_v5_out = loco::must_cast<CircleNonMaxSuppressionV5Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v5_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITOUT:
+    {
+      const auto split_out = loco::must_cast<CircleSplitOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(split_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITVOUT:
+    {
+      const auto splitv_out = loco::must_cast<CircleSplitVOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(splitv_out->input());
+    }
+    case luci::CircleOpcode::CIRCLETOPKV2OUT:
+    {
+      const auto top_kv2_out = loco::must_cast<CircleTopKV2Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(top_kv2_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNIQUEOUT:
+    {
+      const auto unique_out = loco::must_cast<CircleUniqueOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unique_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNPACKOUT:
+    {
+      const auto unpack_out = loco::must_cast<CircleUnpackOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unpack_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEWHILEOUT:
+    {
+      const auto while_out = loco::must_cast<CircleWhileOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(while_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEBIDIRECTIONAL_SEQUENCE_LSTM_OUT:
+      throw std::runtime_error("Unsupported op now");
+    default:
+      return nullptr;
+  }
+}
+
+bool check_layer_info_equal(LayerInfo &left, LayerInfo &right)
+{
+  return left.dtype == right.dtype and left.granularity == right.granularity;
+}
+
+void add_multi_output_node(LayerInfoMap &info_by_name, LayerInfo &layer_info,
+                           luci::CircleNode *node)
+{
+  const auto succs_nodes = loco::succs(node);
+  auto name = node->name();
+
+  if (info_by_name.find(name) != info_by_name.end())
+  {
+    // Check that all outputs have equal dtype and granularity
+    for (const auto succs_node : succs_nodes)
+    {
+      const auto succs_circle_node = loco::must_cast<luci::CircleNode *>(succs_node);
+      name = succs_circle_node->name();
+
+      const auto it = info_by_name.find(name);
+      if (it != info_by_name.end() and not check_layer_info_equal(layer_info, *(it->second)))
+        throw std::runtime_error("Outputs of multiple-output nodes should have equal dtype and "
+                                 "granularity. Check the quantization configuration file");
+    }
+    return;
+  }
+
+  // Add all output nodes to info_by_name
```

Comment:
Is it necessary to add all output nodes? They are ignored in `InsertQuantizeOp`.

Reply:
No, it is unnecessary to add all output nodes. I will remove it.

Reply:
Update: Oh, sorry, I remembered why I added it. I think we need add all output nodes, due to Quantize activation (https://github.com/Samsung/ONE/blob/8366853c8f62aa428bdd85ad44e10f3c3ba9f0cb/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp#L468) part will set correct types to all outputs of multiple output nodes in this case. Otherwise, the Remove redundant Quantize Op pass removes those quantizes, since at this point the output nodes (from multiple output node) types are the same as the output type of the quantizes ops. 
And also I think it is better for QuantizedModelVerifier::verify. So, I prefer add all outputs name. If I'm wrong please correct me :)

Reply:
You're right. Let's add all of them.